### PR TITLE
i/apparmor: pin abi 3.0

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Comment by alexmurray** [comment]
And so it may be easier to just have snapd ship a features file and then invoke apparmor_parser with the `--parser-features` flag to refer to this file rather than hard-coding it in the profiles themselves (as it may not be available on the host system)

File: cmd/snap-confine/snap-confine.apparmor.in [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Author: Jamie Strandboge <jamie@canonical.com>
+abi <abi/3.0>,
```
Comment (zyga) [inline cmd/snap-confine/snap-confine.apparmor.in:2]: This does not work on apparmor_parser 2.10.95 as seen on up-to-date ubuntu core 16

**Comment by zyga** [comment]
The reason we wanted to do it like that was to have consistent behavior when snap apparmor profiles are compiled by distribution apparmor code. I don't have 100% confidence it does not happen anywhere and that at all times we have snapd.apparmor.service doing that work.

**Comment by zyga** [comment]
I can confirm that this does indeed work on core 16:

```sh
apparmor_parser --features-file 3.0 --skip-kernel-load --skip-cache snap-update-ns.hello-world 
```

Where the `3.0` file is literally:
```
query {label {multi_transaction {yes
}
data {yes
}
perms {allow deny audit quiet
}
}
}
dbus {mask {acquire send receive
}
}
signal {mask {hup int quit ill trap abrt bus fpe kill usr1 segv usr2 pipe alrm term stkflt chld cont stop stp ttin ttou urg xcpu xfsz vtalrm prof winch io pwr sys emt lost
}
}
ptrace {mask {read trace
}
}
caps {mask {chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap mac_override mac_admin syslog wake_alarm block_suspend audit_read perfmon bpf
}
}
rlimit {mask {cpu fsize data stack core rss nproc nofile memlock as locks sigpending msgqueue nice rtprio rttime
}
}
capability {0xffffff
}
namespaces {pivot_root {no
}
profile {yes
}
}
mount {mask {mount umount pivot_root
}
}
network {af_unix {yes
}
af_mask {unspec unix inet ax25 ipx appletalk netrom bridge atmpvc x25 inet6 rose netbeui security key netlink packet ash econet atmsvc rds sna irda pppox wanpipe llc ib mpls can tipc bluetooth iucv rxrpc isdn phonet ieee802154 caif alg nfc vsock kcm qipcrtr smc xdp
}
}
network_v8 {af_mask {unspec unix inet ax25 ipx appletalk netrom bridge atmpvc x25 inet6 rose netbeui security key netlink packet ash econet atmsvc rds sna irda pppox wanpipe llc ib mpls can tipc bluetooth iucv rxrpc isdn phonet ieee802154 caif alg nfc vsock kcm qipcrtr smc xdp
}
}
file {mask {create read write exec append mmap_exec link lock
}
}
domain {version {1.2
}
attach_conditions {xattr {yes
}
}
computed_longest_left {yes
}
post_nnp_subset {yes
}
fix_binfmt_elf_mmap {yes
}
stack {yes
}
change_profile {yes
}
change_onexec {yes
}
change_hatv {yes
}
change_hat {yes
}
}
policy {set_load {yes
}
versions {v8 {yes
}
v7 {yes
}
v6 {yes
}
v5 {yes
}
}
}
```

**Comment by zyga** [comment]
I think https://github.com/snapcore/snapd/pull/13711 is the better way of achieving this in a manner that works everywhere.

**Pull request closed without merging**